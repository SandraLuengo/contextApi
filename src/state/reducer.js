import chinesCities from '../fixtures/cities-of-china.json';

const initialState = {
  cities: chinesCities.cities.map(city => ({ ...city, isSelected: false })),
  searching: '',
  isAllChecked: false,
};


const reducer = (state, action) => {
  switch (action.type) {
    case 'SET_CITIES':
      return {
        ...state,
        cities: action.newValue,
      };
    case 'SET_SEARCH':
      return {
        ...state,
        searching: action.newValue,
      };
    case 'SET_IS_ALL_CHECKED':
      return {
        ...state,
        isAllChecked: action.newValue,
      };
    default:
      return state;
  }
};

export { initialState, reducer };

