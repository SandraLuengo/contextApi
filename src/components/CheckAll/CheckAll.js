import React, { useRef, useEffect } from 'react';
import { useGlobalState } from '../../state/useGlobalState';
import { citiesNumber, changeAll } from '../functions';

const CheckAll = ({ className }) => {
  const [{ cities, searching, isAllChecked }, dispatch] = useGlobalState();
  const inputRef = useRef();
  useEffect(() => {
    inputRef.current.checked = isAllChecked;
  }, [isAllChecked]);
  return (
    <div className={className}>
      <input
        className="checkBox"
        type="checkbox"
        ref={inputRef}
        onClick={() => {
          dispatch({ type: 'SET_IS_ALL_CHECKED', newValue: !isAllChecked });
          changeAll(cities, searching, dispatch, !isAllChecked);
        }}
      />
      <p className="numberItems">{citiesNumber(searching, cities)} items</p>
    </div>
  );
};

export default CheckAll;
