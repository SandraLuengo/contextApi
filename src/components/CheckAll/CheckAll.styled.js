import styled from 'styled-components';
import CheckAll from './CheckAll';
import * as style from '../../static/styles/stylesVariables';

export default styled(CheckAll).attrs({})`
  width: 100%;
  height: 5vh;
  display: flex;
  align-items: center;
  padding: 1%;
  border-style: none none solid none;
  border-color: ${style.borderColor};
  border-width: 1.3px;
  .checkBox {
    margin-right: 4%;
  }
  .numberItems {
      font-size:1rem;
      font-weight:bold;
  }
`;
