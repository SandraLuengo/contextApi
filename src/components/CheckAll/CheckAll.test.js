import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { mount } from 'enzyme';
import CheckAll from './CheckAll';
import { Provider } from '../../state/useGlobalState';
import { initialState, reducer } from '../../state/reducer';

Enzyme.configure({ adapter: new Adapter() });

const props = { className: 'test' };

it('CheckAll component renders correctly', () => {
  const tree = renderer
    .create(<Provider reducer={reducer} initialState={initialState}>
      <CheckAll {...props} />
    </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('<CheckAll/>', () => {
  it('should exist checkbox input', () => {
    const wrapper = mount(<Provider reducer={reducer} initialState={initialState}>
      <CheckAll />
    </Provider>);
    expect(wrapper.find('.checkBox').length).toBe(1);
  });

  it('should change search value', () => {
    const wrapper = mount(<Provider reducer={reducer} initialState={initialState}>
      <CheckAll />
    </Provider>);
    wrapper.find('.checkBox').simulate('click', { target: { checked: true } });
  });
});
