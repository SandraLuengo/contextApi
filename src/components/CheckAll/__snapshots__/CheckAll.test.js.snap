// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckAll component renders correctly 1`] = `
<div
  className="test"
>
  <input
    className="checkBox"
    onClick={[Function]}
    type="checkbox"
  />
  <p
    className="numberItems"
  >
    659
     items
  </p>
</div>
`;
